scripting/src/main/java/org/infinispan/scripting/impl/ScriptMetadata.java

```diff
@@ -3,57 +3,83 @@
 import java.io.IOException;
 import java.io.ObjectInput;
 import java.io.ObjectOutput;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Map.Entry;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
 import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.InfinispanCollections;
 import org.infinispan.commons.util.Util;
 import org.infinispan.container.versioning.EntryVersion;
 import org.infinispan.metadata.Metadata;
 
 /**
- * ScriptMetadata.
+ * ScriptMetadata. Holds meta information about a script obtained either implicitly by the script
+ * name and extension, or explictly by its header
  *
  * @author Tristan Tarrant
  * @since 7.2
  */
 public class ScriptMetadata implements Metadata {
+   private final String name;
+   private final ExecutionMode mode;
+   private final String extension;
+   private final Set<String> parameters;
+   private final Optional<String> language;
+   private final Optional<String> role;
+   private final Optional<String> reducer;
+   private final Optional<String> collator;
+   private final Optional<String> combiner;
+
+   ScriptMetadata(String name, Optional<String> language, String extension, ExecutionMode mode, Set<String> parameters,
+         Optional<String> role, Optional<String> reducer, Optional<String> collator, Optional<String> combiner) {
+      this.name = name;
+      this.language = language;
+      this.extension = extension;
+      this.mode = mode;
+      this.parameters = Collections.unmodifiableSet(parameters);
+      this.role = role;
+      this.reducer = reducer;
+      this.collator = collator;
+      this.combiner = combiner;
+   }
 
-   public static enum MetadataProperties {
-      NAME("name"), LANGUAGE("language"), MODE("mode"), REDUCER("reducer"), COLLATOR("collator"), COMBINER("combiner"), EXTENSION("extension");
-
-      private final String s;
+   public Optional<String> language() {
+      return language;
+   }
 
-      MetadataProperties(String s) {
-         this.s = s;
-      }
+   public String extension() {
+      return extension;
+   }
 
-      @Override
-      public String toString() {
-         return s;
-      }
+   public Set<String> parameters() {
```

Comment:
I was thinking that it would be odd if someone adds a script with parameters `[foo, bar]`, and then the script appears in the UI with parameters `[bar, foo]` - even if the order doesn't really matter.